x-pack/test/monitoring_api_integration/scripts/transform_archive.js

```diff
@@ -0,0 +1,82 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+require('../../../../src/setup_node_env');
+
+const { createReadStream, createWriteStream } = require('fs');
+const { mkdir } = require('fs/promises');
+const path = require('path');
+const { Transform } = require('stream');
+const { createGzip } = require('zlib');
+const { createPromiseFromStreams } = require('@kbn/utils');
+
+const {
+  createParseArchiveStreams,
+} = require('../../../../packages/kbn-es-archiver/src/lib/archives/parse'); // eslint-disable-line @kbn/imports/uniform_imports
+
+/**
+ * generates .monitoring-* (metricbeat) archive from a metrics-* (package) archive
+ * options:
+ *   --src <archive-source> - eg /path/to/archive/data.json.gz
+ */
+(async () => {
+  const opts = process.argv.slice(2);
```

Comment:
Just an idea: Could this script also delete the `mappings.json`, in case someone forgets to manually delete it?

Reply:
added removal of the mappings to the script